projects/driver_controls/src/can_fsm.c

```diff
@@ -17,10 +17,10 @@ FSM_STATE_TRANSITION(state_can_transmit) {
 }
 
 static void prv_transmit_data(FSM *fsm, const Event *e, void *context) {
-  InputEventData *data = (InputEventData *)&e->data;
-
-  printf("Device = %d, State = %d, Data = %d\n", e->id, data->components.state,
-         data->components.data);
+  printf("Device = %d, State = %d, Data = %d\n",
+          e->id - 18,
```

Comment:
Sure. I'd also recommend adding a function to actually split the event data - otherwise it isn't clear where these are coming from.